### DISCUSSION TITLE
Play App Signing Upload Certificate

### DISCUSSION THREAD

**Discussion by marchbold** [airsdk/Adobe-Runtime-Support#1209] [Q&A]
Has anyone figured out how to use the "upload certificate" for Play app signing? 

You can download the public `upload_cert.der` from the play console, but struggling to convert this to p12 to use with `adt`. 

I'm just trying to complete the guide on [Packaging AAB](https://airsdk.dev/docs/tutorials/platform/android/packaging-android-app-bundles).

**Comment by sjabberwocky** [comment]
Hi, I use it.
I hardly remember how to generate it, but from my notes in looks like you need to:
1. Generate it by Android Studio.
https://developer.android.com/studio/publish/app-signing.html#generate-key
Key alias should be 'upload'.
2. Convert JSK to P12 for Flash Builder (in my case)
`keytool -importkeystore -srckeystore android_upload.keystore -srcstoretype JKS -deststoretype PKCS12 -destkeystore android_upload.p12`
3. Convert JKS to PEM for Google Play
`keytool -export -rfc -alias upload -file upload_certificate.pem -keystore keystore.jks`

Hope this helps you.

**Comment by leossmith** [comment]
Maybe this will help: https://medium.com/@kanellopoulos.leo/air-apache-flex-2021-how-to-prep-upload-android-app-bundle-f23823ec36b0
